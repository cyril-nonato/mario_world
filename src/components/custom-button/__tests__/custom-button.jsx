import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import CustomButton from '../custom-button.component'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'

describe('Name of the group', () => {

  afterEach(cleanup)

  let wrapper;

  const onClick = jest.fn();
  const type = 'submit'

  beforeEach(() => {
    wrapper = render(<CustomButton onClick={onClick} type={type}>button</CustomButton>)
  })


  test('Should render the correct props', () => {
    const { getByTestId } = wrapper

    expect(getByTestId('button')).toHaveAttribute('type', type);
    expect(getByTestId('button')).toHaveTextContent('button');
  });

  test('Should call the onClick function', () => {
    const { getByTestId } = wrapper;

    fireEvent.click(getByTestId('button'));

    expect(onClick).toHaveBeenCalled();
  });

  test('Should match the snapshot', () => {
    const { asFragment } = wrapper;

    expect(asFragment()).toMatchSnapshot();
  })
});