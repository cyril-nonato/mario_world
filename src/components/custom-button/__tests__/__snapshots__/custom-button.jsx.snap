// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group Should match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: #FFFFFF;
  font-weight: bolder;
  background-color: #454545;
  border: none;
  border-radius: 2px;
  padding: .6rem 1.2rem;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
}

<button
    class="c0"
    data-testid="button"
    type="submit"
  >
    button
  </button>
</DocumentFragment>
`;
